RAGchain/benchmark/dataset/asqa.py

```diff
@@ -0,0 +1,141 @@
+        self.run_pipeline = run_pipeline
+
+        if evaluate_size is not None and len(self.question) > evaluate_size:
+            self.question = self.question[:evaluate_size]
+            self.answers = self.answers[:evaluate_size]
+            self.retrieval_gt = self.content[:evaluate_size]
+
+    def ingest(self, retrievals: List[BaseRetrieval], db: BaseDB, ingest_size: Optional[int] = None):
+        """
+        Ingest dataset to retrievals and db.
+        :param retrievals: The retrievals that you want to ingest.
+        :param db: The db that you want to ingest.
+        :param ingest_size: The number of data to ingest. If None, ingest all data.
+        """
+        ingest_data = deepcopy(self.content)
+
+        if ingest_size is not None:
+            # ingest size must be larger than evaluate size.
+            if ingest_size >= self.eval_size:
+                ingest_data = ingest_data[:ingest_size]
+            else:
+                raise ValueError("ingest size must be same or larger than evaluate size")
+
+        passages = ingest_data.apply(self.__make_passages, axis=1).tolist()
+        passages = list(itertools.chain.from_iterable(passages))
+
+        for retrieval in retrievals:
+            retrieval.ingest(passages)
+        db.create_or_load()
+        db.save(passages)
+
+    def evaluate(self, **kwargs) -> EvaluateResult:
+        question = self.question['ambiguous_question']
+        answer_gt = [answers for answers in self.answers['answer_gt']]
+        retrieval_gt = self.retrieval_gt.apply(self.__make_retrieval_gt, axis=1).tolist()
+
+        return self._calculate_metrics(
+            questions=question,
+            pipeline=self.run_pipeline,
+            retrieval_gt=retrieval_gt,
+            answer_gt=answer_gt,
+            **kwargs
+        )
+
+    def __make_passages(self, row):
+        passage = []
+        for idx, content_dict in enumerate(row['retrieval_gt']):
+            passage.append(Passage(
+                id=str(row['sample_id']) + '_' + str(idx),
+                content=content_dict['content'],
+                filepath=self.file_path,
+                metadata_etc={
+                    'title': content_dict['wikipage'],
+                    'url': ", ".join([wikipages['url'] for wikipages in row['wikipages']]),
+                }
+            ))
+
+        return passage
+
+    def __split_content_answer(self, row):
+        content_lst = []
+        answer_lst = []
+        for element in row['annotations']:
+            if len(element['knowledge']) != 0:
+                content_lst += [content for content in element['knowledge']]
+            answer_lst += [element['long_answer']]
+
+        if len(content_lst) == 0 or len(answer_lst) == 0:
+            row['retrieval_gt'], row['answers'] = np.NAN, np.NAN
+        else:
+            row['retrieval_gt'], row['answers'] = content_lst, answer_lst
+
+        return row['retrieval_gt'], row['answers']
```

Comment:
just return np.nan, np.nan when len(content_lst) == 0 or len(answer_lst) == 0
else return content_lst, answer_lst directly?
Why need to assign row['retrieval_gt'] and row['answers'] in this function? You will use this to apply and assign to new columns anyway..